apis/risingwave/v1alpha1/risingwave_state_store_backend.go

```diff
@@ -202,6 +218,25 @@ type RisingWaveStateStoreBackendAzureBlob struct {
 	Endpoint string `json:"endpoint"`
 }
 
+// RisingWaveStateStoreBackendAliyunOSS is the details of AliyunOSS for compute and compactor components.
+type RisingWaveStateStoreBackendAliyunOSS struct {
+	// RisingWaveAliyunOSSCredentials is the credentials provider from a Secret.
+	RisingWaveAliyunOSSCredentials `json:"credentials"`
+
+	// Bucket name of your AliyunOSS
+	// +kubebuilder:validation:Required
+	Bucket string `json:"bucket"`
+
+	// Working directory root of the Aliyun OSS.
+	// +kubebuilder:validation:Required
+	Root string `json:"root"`
+
+	// Endpoint of the Aliyun OSS.
+	// e.g. https://oss-cn-hangzhou.aliyuncs.com
+	// +kubebuilder:validation:Pattern="^(?:https://)?(?:[^/.\\s]+\\.)*(?:[^/\\s]+)*$"
+	Endpoint string `json:"endpoint"`
```

Comment:
Unused field.

Reply:
It is not used in this PR, but In current main branch, we use the `Endpoint`.
https://github.com/risingwavelabs/risingwave-operator/blob/main/pkg/factory/risingwave_object_factory.go#L713C41-L713C41

Reply:
I see. But the endpoint can be deduced from other parameters. Can we keep it compatible as before?

https://github.com/risingwavelabs/risingwave-operator/blob/ae8b883ab600cec87debb183c5ebf01a92a6827f/pkg/factory/risingwave_object_factory.go#L56-L57

Reply:
Ok, then we need to add the `Region` and `InternalEndpoint` parameter, right?

Reply:
I am not familiar with Aliyun. Which way will be easier for user to use?

Reply:
I think it's the original way. Users should be bothered with endpoints.